src/Test/Utilities/Portable/Diagnostics/OptionsDiagnosticAnalyzer.cs

```diff
@@ -16,6 +17,7 @@ public class OptionsDiagnosticAnalyzer<TLanguageKindEnum> : TestDiagnosticAnalyz
         public OptionsDiagnosticAnalyzer(AnalyzerOptions expectedOptions)
         {
             _expectedOptions = expectedOptions;
+            Debug.Assert(expectedOptions.PerTreeOptions.GetType() == typeof(CompilerPerTreeOptionsProvider));
```

Comment:
This feels wrong. Overall this seems to be a pervasive pattern in our options code now: having APIs that accept public types but then asserting / throwing if it's not a specific derived type. Makes me question if we have the appropriate design here.

Reply:
This is a test utility class. I'm asserting that we're always using the internal compiler provider in our tests.